Library/Homebrew/test/cask/system_command_spec.rb

```diff
@@ -1,4 +1,65 @@
 describe Hbc::SystemCommand, :cask do
+  describe "#initialize" do
+    let(:env_args) { ["bash", "-c", 'printf "%s" "${A?}" "${B?}" "${C?}"'] }
+
+    describe "given some environment variables" do
+      subject {
+        described_class.new(
+          "env",
+          args: env_args,
+          env: { "A" => "1", "B" => "2", "C" => "3" },
+          must_succeed: true,
+        )
+      }
+
+      its("run!.stdout") { is_expected.to eq("123") }
+
+      describe "the resulting command line" do
+        it "does not include the given variables" do
+          expect(Open3)
```

Comment:
I think this should work:

```ruby
expect(Open3).to receive(:popen3).with(a_hash_including("PATH"), ["env"] * 2, *env_args, {})
```

Reply:
This breaks for me, apparently because it causes the test double to return `nil` to the caller (instead of the expected `stdin, stdout, stderr, wait_thr` tuple.)

Reply:
Maybe append `.and_call_original`?

Reply:
My rationale behind `/usr/bin/true` was to avoid spawning a `bash` we don’t need.

But then again, it’s premature optimization, and maybe you’re right.

Reply:
Thanks again for your feedback @reitermarkus! I have addressed it with 071152b.